src/login/oidc/oidcHandlers/LegacyImplicitFlowOidcHandler.ts

```diff
@@ -60,11 +69,21 @@ export default class LegacyImplicitFlowOidcHandler implements IOidcHandler {
     const query: Record<string, string> = {
       client_id: oidcLoginOptions.client.clientId,
       response_type: "id_token token",
-      redirect_url: oidcLoginOptions.redirectUrl.toString(),
+      redirect_uri: oidcLoginOptions.redirectUrl.toString(),
       // The webid scope does not appear in the spec
       scope: "openid webid offline_access",
       state: oidcLoginOptions.sessionId
     };
+
+    await this.dpopClientKeyManager.generateClientKeyIfNotAlready();
+    await this.storageUtility.setForUser(
+      oidcLoginOptions.sessionId,
```

Comment:
Run these in parallel?

Reply:
Fixed in e65ae11bb5e81297c1fbc3e19e0219f4d3b5282d